### ISSUE TITLE
Turing Institute joining the BinderHub Federation

### ISSUE THREAD

**Issue by sgibson91** [jupyterhub/mybinder.org-deploy#1124]
Hi all 👋 

We've talked for a while about the Turing joining the BinderHub Federation. This issue to document the status of the process.

### Step 1: @sgibson91 submits an Azure credit request to the Turing Institute

HackMD: https://hackmd.io/SxF25aeRRCS1ZptyShYxtg?view

The application takes the following form:

- A project title
- The project PI's name and email
- Whether there is an existing subscription
- The amount of credits requested
- Start and end dates for the allocation
- A platform justification
- A research justification (A description of the project and the research outputs the Azure credits will support (e.g. papers, datasets, codebase etc.))
- The computational requirements
- A list of users who require access and their emails (Can include collaborators)
- A Costing breakdown

The plan is request £10k to eventually support ~20% of the mybinder.org traffic. I suggest making this request at the beginning of October as that is when the financial year resets. I will also recommend that the project be capped at £10k/year so we can control our spending and not accept any traffic during periods when our subscription is deactivated.

If anyone has any thoughts/opinions on what would make this a knock-out proposal (especially the research justification section), please let me know! I'll set up a shared doc soon. ✨

cc: @KirstieJane @martintoreilly

### Updates

**2019-08-29:** Credit request submitted

**Comment by choldgraf** [comment]
I am more than happy to help put together the proposal (or to distance myself from it and then write a letter of support)...whatever helps the most.

I think this would be fantastic!

**Comment by sgibson91** [comment]
Thanks @choldgraf! I think it'll be all hands on deck as I don't think an external letter of support has ever been submitted for this process (I'll double check though, it's gone through some changes).

**Comment by betatim** [comment]
For the research justification: do we know GitHub users or organisations that the Turing cares about? Then we could look at https://archive.analytics.mybinder.org/ to fish out some relevant examples of papers or workshops that mybinder.org has hosted so far. Then solicit a testimonial from those people?

The event archive is great both in how cool it is and in size. I find it hard to find stuff without having an idea for what I am searching for. There is just too much stuff.

**Comment by sgibson91** [comment]
I don't know about specific users/orgs, but I think showing that mybinder supports open science and best practices (because you have to be following open science principals and best practices in order to use mybinder) would be enough. I think we should definitely link out to the event archive to show how widely used mybinder is, and we could maybe pick out a couple of examples we think are particularly cool?

**Comment by KirstieJane** [comment]
@sgibson91 - would you mind if we spent 10 minutes of our meeting about your talk going through this application? I think we're in a great position to justify this application.

I would really like to be PI on the project if that's ok, but happy to keep it in REG if that makes more sense?

**Comment by sgibson91** [comment]
@KirstieJane great idea! I'm happy for you to be PI as well 😄

**Comment by sgibson91** [comment]
Here's a hackmd with some _very_ rough notes! https://hackmd.io/SxF25aeRRCS1ZptyShYxtg?view

**Comment by betatim** [comment]
Note to Tim: CMS Open Data repo as example

**Comment by sgibson91** [comment]
@KirstieJane In the Binder meeting this morning, we were wondering if there are any examples of Binderised repos that produced papers for the research justification section. I wondered if you had any ideas?

Also, would a Binder Operator be interested in being a collaborator on the Azure subscription. This would allow Azure knowledge relating to operating a Binder propagate through the team as well as provide a safety net so I'm not the sole failure point for the the cluster (especially since I'm going on holiday for a month at the end of October!). @betatim or @minrk would be the obvious choices for timezone reasons, but I'm open to anyone from the team who has an interest! 😊

**Comment by betatim** [comment]
I'd be happy to get access and learn about azure. It also makes it easier to fix stuff if we don't have to email someone to do stuff.

**Comment by sgibson91** [comment]
Awesome! Unless there's a continued discussion on this, I'll write Tim into the request 🚀

**Comment by choldgraf** [comment]
I'm more than +1 on Tim helping with this - though I think we should also in general try to find new people (or people who aren't already helping with another binderhub deployment) to help with these kinds of deployments, rather than carve off percentages of pre-existing people. Obviously there's a question there around sustainability and why they'd justify their time working on this, but I think it's something to strive for in general :-)

**Comment by sgibson91** [comment]
Agreed. It'd be great to get another member of Turing REG on-board, for instance - I'm just not sure how without a formal project.

**Comment by sgibson91** [comment]
I've done some cost analysis for Azure resources.

```
# Container Registry
# Premium SKU: 500GB storage included
#====================================
acr_price_perday = $1.667
acr_price_peryear = price_perday * 365.25

# Costs for extra storage beyond included 500GB
# Based on 2TB of OVH cluster
storage_price_perday_perGB = $0.003
extraGB = (2 - 0.5) * 1000
storage_price_peryear = storage_price_perday_perGB * extraGB * 365.25

total_acr_price = acr_price_peryear + storage_price_peryear
                = $2252.50

# K8s cluster VMs
# General Purpose D4s v3: 4 vCPUs, 16GB RAM
#==========================================
vm_price_permonth = $175.50
cluster_price_permonth = 3 * vm_price_permonth
cluster_price_peryear = 12 * cluster_price_permonth
                      = $6318.00

# Burstable VM: B1S - to run automated clean up operations
#=========================================================
vm_price_permonth = $8.76
vm_price_peryear = 12 * vm_price_permonth
                 = $105.12

# Total Budget
#=============
total_price_peryear = total_acr_price + cluster_price_peryear + vm_price_peryear
                    = $8675.62
```

**Comment by choldgraf** [comment]
Just a note to add to Sarah's point: right now I believe OVH is at 15% of Binder traffic.

**Comment by choldgraf** [comment]
Another note here - I wonder if this is something that we could get more Azure credits for, particularly if we've already got folks willing to run it on Azure

**Comment by betatim** [comment]
The way I understood it is that Sarah isn't looking for at most one person, but at least one person. I think the more the merrier (up to a point :-)).

**Comment by betatim** [comment]
> Another note here - I wonder if this is something that we could get more Azure credits for directly from Microsoft, particularly if we've already got folks willing to run it on Azure

Nods. I think we can have a Turing hub and in parallel have a Azure hub in the federation. Who gives credits for the Turing hub is what programmers would call "an implementation detail".

**Comment by manics** [comment]
> The way I understood it is that Sarah isn't looking for at most one person, but at least one person.

I'm happy to help out, depending on what the expectations are! Don't have much experience (yet) with Binderhub but I've run Jupyterhub on AKS before- I updated the Z2JH docs last year after obtaining some free Azure credits.

@sgibson91 I'll be at RSECon so could discuss then?

**Comment by betatim** [comment]
Step one (independent of Turing hub) is get yourself involved in running mybinder.org and the software behind it. It would be super cool to have you onboard!

**Comment by sgibson91** [comment]
@manics Let's definitely chat at RSE Con! And I agree with @betatim, in the meantime you could dip your toes in to managing mybinder.org in its current guise

**Comment by choldgraf** [comment]
Hey all - friendly ping on this! @sgibson91 how is the proposal going? Let us know if we can help in particular ways!

**Comment by sgibson91** [comment]
I've been at a conference this week so slow going unfortunately. I have a meeting with @KirstieJane on Wednesday where we're going to chat about this. I think the Research Justification is probably where you all can help out most. A couple of sentences is probably fine - it looks a little more formal than it actually is 😂

**Comment by sgibson91** [comment]
Also, the costing breakdown I laid out in https://github.com/jupyterhub/mybinder.org-deploy/issues/1124#issuecomment-521756497 is about $1500 under the budget of $10k. Should we reduce what we ask for to make it match up with the breakdown, or try to justify by including autoscaling and try to predict how much autoscaling we'd actually get?

**Comment by betatim** [comment]
I'd keep it at a "nice round" 10k and add a sentence "we leave a bit of head room for experimenting with auto-scaling to make the hub more cost effective" or some such.

**Comment by choldgraf** [comment]
I agree a cool 10K is good, that way you have a bit of wiggle room to figure things out. I suspect that this will be a learning experience, since deploying BinderHub within the federation will result in it getting a lot more traffic, and it require some refining of the approach that may result in a bit more spend than usual.

---

A spit-balled research statement:

The BinderHub Federation is an experiment in open infrastructure for open science. It is a community of scientists and open source developers dedicated to scientific reproducibility, communicating ideas through interactive computing, and providing access to results and analyses. The BinderHub Federation provides interactive data analytics environments to nearly 300,000 people a month, and it relies on its team of partner BinderHubs to collaborate in providing this service. These research credits allow the Turing Institute to be one of the core members of the BinderHub Federation, and allow us to be a leader in growing this cutting-edge community of open science advocates.

---

It's a little bit bullshitty, so feel free to tone that down if you like :-)

**Comment by betatim** [comment]
I really like the first sentence! We should use it more!

I would edit the "300,000 people" because it could be understood as "300k different people use mybinder.org each month to do stuff". For all we know there are 10people in the world who launch a hell of a lot of binders ;) I think it is fair to write "Around 300k Binders are launched every month from around 10k unique repositories."

I think we have some impressive stats so we should be precise and set ourselves apart from those who can't be precise (and impressive) because they don't have the facts ;)

If you want to you could add/weave in at the end something about "Binder is an established but still young project. By doing this now we aren't investing in unproven tech/team w/o a track record while still being early enough to the party that we can shape where things will go. We are leading/shaping the future of scientific knowledge sharing." (needs a language polish)

**Comment by sgibson91** [comment]
@betatim @choldgraf I attempted to morph your statements/suggestions in the hackmd: https://hackmd.io/SxF25aeRRCS1ZptyShYxtg?view#Research-Justification

**Comment by betatim** [comment]
LGTM

**Comment by KirstieJane** [comment]
Heya! I've given the research justification a bit of Turing specific oomph. Please have a read through (in the hackmd) and let me know if anything I say is wrong or way over the top 😆 

Thank you for all the great work on this - I'm delighted that we're moving towards joining the federation!! 💖  

![](https://media.giphy.com/media/1sv8qbOdO9d5mxnNEX/giphy.gif)

**Comment by sgibson91** [comment]
Love it @KirstieJane, and the suggested title! :sparkles:

**Comment by betatim** [comment]
Submit it now :) I am happy with all of it and also like the suggested title.

**Comment by sgibson91** [comment]
I formulated the "Platform Justification" and "Computational Requirements" into fuller sentences and I think we're ready to submit! :tada:

**Comment by sgibson91** [comment]
@KirstieJane and I did some more cleaning up this morning and I've just submitted the request!

**Comment by betatim** [comment]
Closing this then! Good work everyone and let's make a new issue if the request gets approved.

**Comment by choldgraf** [comment]
wahooo! good job getting it 🚢 'ed, and good luck getting the funding! If it doesn't come through lets brainstorm some other ways to get y'all into the federation :-)

**Comment by meeseeksmachine** [comment]
This issue has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/how-can-i-support-mybinder-org/4692/4
